### PR TITLE
Add domain validation to map publishing

### PR DESCRIPTION
**Pull request by mikkomustikkamaa**
Added domain validation to map publishing using regular expression available at https://miguelmota.com/bytes/validate-domain-regex/. Yellow warning icon with black text is shown to user with 1 second delay when needed.

### CODE REVIEW

File: bundles/framework/publisher2/view/PanelGeneralInfo.js [hunk 1]
```diff
@@ -58,6 +59,24 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PanelGeneralInfo
                     field.setLabel(data.label);
                     field.setTooltip(data.tooltip, data.helptags);
                     field.setPlaceholder(data.placeholder);
+
+                    const keyUpHandler = Oskari.util.throttle((event) => {
```
Comment (ZakarFin) [inline bundles/framework/publisher2/view/PanelGeneralInfo.js]: Define the function outside the loop and rename it to something like domainValidatorKeyUpHandler as it's only used for domain.

File: bundles/framework/publisher2/view/PanelGeneralInfo.js [hunk 2]
```diff
@@ -125,6 +144,16 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PanelGeneralInfo
 
             me._languageChanged(selectedLang);
         },
+        /*
+         * Function implemented by modifying function introduced in https://miguelmota.com/bytes/validate-domain-regex/
+         */
+        _isValidDomain (v) {
```
Comment (ZakarFin) [inline bundles/framework/publisher2/view/PanelGeneralInfo.js]: This should be moved to Oskari.util.

File: bundles/framework/publisher2/view/PanelGeneralInfo.js (hunk 1 above)
Comment (mikkomustikkamaa) [reply bundles/framework/publisher2/view/PanelGeneralInfo.js]: Fixed as proposed.

File: bundles/framework/publisher2/view/PanelGeneralInfo.js (hunk 2 above)
Comment (mikkomustikkamaa) [reply bundles/framework/publisher2/view/PanelGeneralInfo.js]: Function moved to Oskari.util and some unit tests is added.

**Comment by ZakarFin** [comment]
Adds Oskari.util.isValidDomain("domain as string")

**Pull request merged**